### ISSUE TITLE
DEBUG_DEPTH not working

### ISSUE THREAD

**Issue by anishkny** [debug-js/debug#746]
`DEBUG_DEPTH` env var has no effect. I tried values of `10`, `100`, `null`. No matter what I always get the following display:

```
2020-02-08T22:17:01.423Z SCOPE STRING [ { email: 'xxx@yyy.com', vars: [ [Object] ] } ]
```

My invocation: 
```
debug('STRING', targetObj);
```

**Comment by BlackSparrowz** [comment]
GitHub Enterprise Server only: This is all possible on GitHub Enterprise through GitHub Connect. It may take up to an hour to refresh the alerts and make them visible. After waiting a reasonable amount of time, if you are still not seeing the yellow bar in the Dependency Graph, you may want to contact your administrator. In the mean time, to move along with the course, we'll give you a hint - the recommended upgraded version is 2.6.9

**Comment by chriskinsman** [comment]
Noticed this also.  Looks like it is defaulting to 3 and not allowing an override...

Any fix in the works?

**Comment by Qix-** [comment]
Yep looks like a regression somewhere.

**Comment by Qix-** [comment]
Can you test with `npm i --no-save visionmedia/debug#fix/depth`?

**Comment by Qix-** [comment]
Started down the `git blame` rabbit hole. This has been broken for years. Amazing.

**Comment by anishkny** [comment]
Looks like fix is effective. Any place to add unit tests for this?

I should have noted it reproduces only when deep object is the _second_ argument to the `debug` function, e.g.
```js
// index.js
require('debug')('FOO')('something', {
  x: 1,
  y: { a: 2, b: 3, c: { a: 1, b: { a: 1, b: { a: 1, b: { a: 1 } } } } },
});
```

Before fix:
```bash
$ DEBUG=FOO DEBUG_DEPTH=40 node index.js
  FOO something { x: 1, y: { a: 2, b: 3, c: { a: 1, b: [Object] } } } +0ms
```

After fix:
```bash
$ DEBUG=FOO DEBUG_DEPTH=40 node index.js
  FOO something {
  x: 1,
  y: {
    a: 2,
    b: 3,
    c: {
      a: 1,
      b: { a: 1, b: { a: 1, b: { a: 1 } } }
    }
  }
} +0ms
```

**Comment by Qix-** [comment]
> I should have noted it reproduces only when deep object is the second argument to the debug function, e.g.

Yeah that's much less simple to fix; there are a few bugs about this. v5 is going to be a rewrite for the most part, so a lot of this will get addressed then.

Glad to know it works now; I'll commit it in. Thank you very much for testing @anishkny :)

**Comment by sylver** [comment]
What is the status on this issue ?

I noticed it recently in our production container, but that's a bit more tricky since I can't reproduce the problem locally.

Same value of `DEBUG_DEPTH` in both case (`10`).
Trying to decrement or increment it locally seems to produce the expected effect, while any value in our container does nothing (looks like it's stucked to `3`), so I'm guessing something else is interfering but I wasn't able to find out what yet.

**Comment by Qix-** [comment]
> What is the status on this issue?

![image](https://user-images.githubusercontent.com/885648/119178060-80339d80-ba6d-11eb-8f88-37ce6c8639e8.png)

**Comment by sylver** [comment]
@Qix- very useful & constructive, thanks.

Since you clearly need me to clarify my question:

Giving this issue is opened for more than a year, and no news since Jul 9, 2020, would you be so kind to provide some update on its resolution ?

Thanks

**Comment by Qix-** [comment]
There is no update.

**Comment by sylver** [comment]
@Qix- Thanks.

Thing is, you apparently already had a fix a year ago and stated in the last update on this issue :

>  [...] I'll commit it in sometime soon.

I'm wondering what's holding you back for so long and if the project is still maintained.
Do you or any maintainer on the project would need any help to make it happen ?

**Comment by Qix-** [comment]
@sylver

> Thing is, you apparently already had a fix a year ago and stated in the last update on this issue :

Yep.

> I'm wondering what's holding you back for so long

Life.

> if the project is still maintained.

Yep.

> Do you or any maintainer on the project would need any help to make it happen ?

I'm the only active maintainer and I've been struggling to find free time to work on unrelated OSS. The pandemic hit us really hard and we're recovering still.

**Comment by sylver** [comment]
@Qix- 

> The pandemic hit us really hard and we're recovering still.

Sorry to hear that, hold on there.

> I'm the only active maintainer and I've been struggling to find free time to work on unrelated OSS.

That's understandable.

That's not a critical issue anyway, just an annoying one.   
But if there's a quick win and I can help to close it with a MR (adding what's missing, unit tests maybe ?), let me know.

**Comment by calvintwr** [comment]
@Qix- I have created a PR https://github.com/debug-js/debug/pull/911 for this fix.

It comes with tests as well.

**Comment by eavdmeer** [comment]
It looks like your fix unfixed itself in the mean time somewhere. Running the test someone mentions above yields this:
```
DEBUG=FOO DEBUG_DEPTH=40 node bla.js
  FOO something { x: 1, y: { a: 2, b: 3, c: { a: 1, b: [Object] } } } +0ms
```
As you can see, this stil (again?) prints `[Object]`

**Comment by calvintwr** [comment]
> It looks like your fix unfixed itself in the mean time somewhere. Running the test someone mentions above yields this:
> 
> ```
> DEBUG=FOO DEBUG_DEPTH=40 node bla.js
>   FOO something { x: 1, y: { a: 2, b: 3, c: { a: 1, b: [Object] } } } +0ms
> ```
> 
> As you can see, this stil (again?) prints `[Object]`

My fix wasn't merged. You can try to checkout this branch to see if it works: https://github.com/calvintwr/debug/tree/fix/debug-depth

**Comment by eavdmeer** [comment]
Looks like it works as expected now:
```
$ DEBUG=FOO DEBUG_DEPTH=40 node index.js
  FOO something {
  x: 1,
  y: {
    a: 2,
    b: 3,
    c: {
      a: 1,
      b: { a: 1, b: { a: 1, b: { a: 1 } } }
    }
  }
} +0ms
$ DEBUG=FOO DEBUG_DEPTH=1 node index.js
  FOO something { x: 1, y: { a: 2, b: 3, c: [Object] } } +0ms
```

**Comment by piboistudios** [comment]
@Qix now that the pandemic is over you think you could merge this? :)

**Comment by calvintwr** [comment]
@7elephants this project is abandoned. We are maintaining it at https://github.com/calvintwr/debug-next

**Comment by eavdmeer** [comment]
Abandoned at 200M weekly download versus 80 for debug-next???

**Comment by calvintwr** [comment]
@7elephants the original debug team abandoned this project. We are not associated with them. We forked the repo and maintain it separately as a drop in replacement, with bug fixes and extra features that we needed.

**Comment by Qix-** [comment]
@calvintwr please stop spreading misinfo. It's not abandoned. You're welcome to fork but it's not entirely easy to maintain this package as I'd like to.

**Comment by Qix-** [comment]
Released as `4.3.5`.

**Comment by eavdmeer** [comment]
@Qix- Thanks for the update. That debug-next fork is interesting, but I strongly prefer debug as it keeps things exactly as they should be: simple!
